plugins/logging/pkg/backend/metadata.go

```diff
@@ -0,0 +1,269 @@
+package backend
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchutil"
+	opnicorev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/capabilities/wellknown"
+	"github.com/rancher/opni/pkg/opensearch/opensearch/types"
+	"github.com/rancher/opni/pkg/resources/multiclusterrolebinding"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"net/http"
+	"os"
+	"time"
+)
+
+func (b *LoggingBackend) waitForOpensearchClient(ctx context.Context) error {
+	var retErr error
```

Comment:
is there any reason not to use `b.WaitForInit()` here?  If we want a method that handles a context closing I think it would be better off adding it in there.

Reply:
Not really. The only reason we might want to avoid it here is if we have to add more init code that relies on the opensearch client and need to move the call to `waitForOpensearchClient` outside of the goroutine [here](https://github.com/rancher/opni/blob/human-logging/plugins/logging/pkg/backend/logging.go#L63-L75) which would block `LoggingBackend.Initialize(...)` indefinitely. But this is super unlikely. I'll add `b.WaitForInit()`